modules/tinymce/src/plugins/searchreplace/test/ts/module/test/Utils.ts

```diff
@@ -1,64 +1,59 @@
-import { Assertions, Chain, GeneralSteps, Guard, UiControls, UiFinder, Waiter } from '@ephox/agar';
-import { TinyUi } from '@ephox/mcagar';
+import { UiControls, UiFinder, Waiter } from '@ephox/agar';
+import { TinyUiActions } from '@ephox/mcagar';
 import { SugarElement } from '@ephox/sugar';
 
-const cFakeEvent = (name: string) => Chain.label('Fake event',
-  Chain.op((elm: SugarElement) => {
-    const evt = document.createEvent('HTMLEvents');
-    evt.initEvent(name, true, true);
-    elm.dom.dispatchEvent(evt);
-  })
-);
+import Editor from 'tinymce/core/api/Editor';
 
-const cFindInDialog = (ui: TinyUi, selector: string) => Chain.control(
-  Chain.fromChains([
-    ui.cWaitForPopup('Wait for dialog', 'div[role="dialog"]'),
-    UiFinder.cFindIn(selector)
-  ]),
-  Guard.addLogging(`Find ${selector} in dialog`)
-);
+const fakeEvent = (elm: SugarElement<HTMLElement>, name: string) => {
+  const evt = document.createEvent('HTMLEvents');
+  evt.initEvent(name, true, true);
+  elm.dom.dispatchEvent(evt);
+};
 
-const sAssertFieldValue = (ui: TinyUi, selector: string, value: string) => Waiter.sTryUntil(`Wait for new ${selector} value`,
-  Chain.asStep({}, [
-    cFindInDialog(ui, selector),
-    UiControls.cGetValue,
-    Assertions.cAssertEq(`Assert ${value} value`, value)
-  ]), 20, 3000
-);
+const pFindInDialog = async (editor: Editor, selector: string) => {
+  const dialog = await TinyUiActions.pWaitForDialog(editor, 'div[role="dialog"]');
+  return UiFinder.findIn(dialog, selector);
```

Comment:
This would likely be better if it called `getOrDie` instead of having to call it wherever it's used.